BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ReflectionCredentialEnvelope.cs

```diff
@@ -0,0 +1,244 @@
+    /// This class uses reflection to interact with the Azure.Core library.
+    /// </summary>
+    /// <remarks>
+    /// Our SDK currently targets net452, net46, and netstandard2.0.
+    /// Azure.Core.TokenCredential is only available for netstandard2.0.
+    /// </remarks>
+    internal class ReflectionCredentialEnvelope : ICredentialEnvelope
+    {
+        private readonly object tokenCredential;
+        private readonly object tokenRequestContext;
+
+        /// <summary>
+        /// Create an instance of <see cref="ReflectionCredentialEnvelope"/>.
+        /// </summary>
+        /// <param name="tokenCredential">An instance of Azure.Core.TokenCredential.</param>
+        public ReflectionCredentialEnvelope(object tokenCredential)
+        {
+            this.tokenCredential = tokenCredential ?? throw new ArgumentNullException(nameof(tokenCredential));
+
+            if (tokenCredential.GetType().IsSubclassOf(Type.GetType("Azure.Core.TokenCredential, Azure.Core")))
+            {
+                this.tokenRequestContext = AzureCore.MakeTokenRequestContext(scopes: CredentialConstants.GetScopes());
+            }
+            else
+            {
+                throw new ArgumentException($"The provided {nameof(tokenCredential)} must inherit Azure.Core.TokenCredential", nameof(tokenCredential));
+            }
+        }
+
+        /// <summary>
+        /// Gets the TokenCredential instance held by this class.
+        /// </summary>
+        public object Credential => this.tokenCredential;
+
+        /// <summary>
+        /// Gets an Azure.Core.AccessToken.
+        /// </summary>
+        /// <param name="cancellationToken">The System.Threading.CancellationToken to use.</param>
+        /// <returns>A valid Azure.Core.AccessToken.</returns>
+        public string GetToken(CancellationToken cancellationToken = default)
+        {
+            SdkInternalOperationsMonitor.Enter();
+            try
+            {
+                return AzureCore.InvokeGetToken(this.tokenCredential, this.tokenRequestContext, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                CoreEventSource.Log.FailedToGetToken(ex.ToInvariantString());
+                return null;
+            }
+            finally
+            {
+                SdkInternalOperationsMonitor.Exit();
+            }
+        }
+
+        /// <summary>
+        /// Gets an Azure.Core.AccessToken.
+        /// </summary>
+        /// <param name="cancellationToken">The System.Threading.CancellationToken to use.</param>
+        /// <returns>A valid Azure.Core.AccessToken.</returns>
+        public async Task<string> GetTokenAsync(CancellationToken cancellationToken = default)
+        {
+            SdkInternalOperationsMonitor.Enter();
+            try
+            {
+                return await AzureCore.InvokeGetTokenAsync(this.tokenCredential, this.tokenRequestContext, cancellationToken).ConfigureAwait(true);
```

Comment:
@cijothomas , Is `ConfigureAwait(true)` the correct usage for `SdkInternalOperationsMonitor` ?

Reply:
I think I answered my own question.
`ProfileServiceWrapper` calls Async methods inside an InternalOperation.
https://github.com/microsoft/ApplicationInsights-dotnet/blob/e1ff87e5e96daf4517ee9c28415f4024ec29529f/BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/ApplicationId/ProfileServiceWrapper.cs#L86-L107